bin/pkg.ml

```diff
@@ -270,15 +264,42 @@ module Lock = struct
 
   let term =
     let+ (common : Common.t) = Common.term
-    and+ env_source = Env.Source.term
     and+ opam_repository_path = Opam_repository_path.term
     and+ opam_repository_url = Opam_repository_url.term
     and+ context_name = context_term
     and+ all_contexts =
       Arg.(
         value & flag
         & info [ "all-contexts" ] ~doc:"Generate the lockdir for all contexts")
-    and+ version_preference = Version_preference.term in
+    and+ version_preference = Version_preference.term
+    and+ use_env_from_current_system =
+      Arg.(
+        value & flag
+        & info
+            [ "use-env-from-current-system" ]
+            ~doc:
+              "Set opam system environment variables based on the current \
+               system when solving dependencies. This will restrict packages \
+               to just those which can be installed on the current system. \
+               Note that this will mean that the generated lockdir may not be \
+               compatible with other systems. If the build context(s) specify \
+               system environment variables then the solver will use the union \
+               of the environment variables infered from the current system \
+               and the environment variables set in the build context(s). If \
+               the current system and the build context(s) disagree on the \
+               value of an environment variable then an error is thrown.")
+    and+ just_print_solver_env =
+      Arg.(
+        value & flag
+        & info
+            [ "just-print-solver-env" ]
+            ~doc:
+              "Print an s-expression describing the environment that would be \
+               used to solve dependencies and then exit without attempting to \
+               solve the dependencies or generate the lockfile. Intended to be \
+               used to debug situations where no solution can be found to a \
+               project's dependencies.")
+    in
```

Comment:
I'm imagining the situation where a user is confused because `dune pkg lock` is producing unexpected results and so they run `dune pkg lock --help`. Since this flag is to help users debug this specific scenario I think it makes sense for it to be discoverable via `dune pkg lock --help` and I think making it a flag of `dune pkg lock` is the simplest way to achieve this.

Reply:
I agree with @gridbugs here. In the future when pkg support is more fleshed out, we can look at adding such a feature to `show`. But for now the indirection doesn't seem to be worth it.